### ISSUE TITLE
Adapters fail serialization if they return null

### ISSUE THREAD

**Issue by dwaite** [eclipse-ee4j/yasson#135]
I created the following adapter for my type, both to convert a duration to milliseconds and to collapse a 'default' duration of 0 to null. This was a second attempt; the first attempt adapted to `Long`.

```java
	class DurationMillisecondsAdapter implements JsonbAdapter<Duration, JsonValue> {

		@Override
		public JsonValue adaptToJson(Duration obj) throws Exception {
			return Duration.ZERO.equals(obj) ? null : Json.createValue(obj.toMillis());
		}

		@Override
		public Duration adaptFromJson(JsonValue obj) throws Exception {
			return obj != null ? Duration.ofMillis(((JsonNumber)obj).longValue()) : Duration.ZERO;
		}
	}
```

This throws:
```
Jun 07, 2018 2:42:27 AM org.eclipse.yasson.internal.Marshaller marshall
SEVERE: Problem adapting object of type class java.time.Duration to interface javax.json.JsonValue in class class com.pingidentity.labs.webauthn.JsonUtils$DurationMillisecondsAdapter
Jun 07, 2018 2:42:27 AM org.eclipse.yasson.internal.Marshaller marshall
SEVERE: Generating incomplete JSON
Exception in thread "main" javax.json.bind.JsonbException: Problem adapting object of type class java.time.Duration to interface javax.json.JsonValue in class class DurationMillisecondsAdapter
	at org.eclipse.yasson.internal.serializer.AdaptedObjectSerializer.serialize(AdaptedObjectSerializer.java:93)
	at org.eclipse.yasson.internal.serializer.AbstractContainerSerializer.serializerCaptor(AbstractContainerSerializer.java:96)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.marshallProperty(ObjectSerializer.java:94)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.serializeInternal(ObjectSerializer.java:63)
	at org.eclipse.yasson.internal.serializer.AbstractContainerSerializer.serialize(AbstractContainerSerializer.java:64)
	at org.eclipse.yasson.internal.Marshaller.serializeRoot(Marshaller.java:121)
	at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:77)
	at org.eclipse.yasson.internal.JsonBinding.toJson(JsonBinding.java:98)
Caused by: java.lang.NullPointerException
	at org.eclipse.yasson.internal.serializer.AdaptedObjectSerializer.resolveSerializer(AdaptedObjectSerializer.java:101)
	at org.eclipse.yasson.internal.serializer.AdaptedObjectSerializer.serialize(AdaptedObjectSerializer.java:90)
	... 9 more
```

For now, I am working around this by building a custom serializer/deserializer for the outer type.

**Comment by Verdent** [comment]
Hi, could you please provide reproducer for this issue?

**Comment by scottkurz** [comment]
This looks similar to the issue I hit.

I wrote it up too as a question on [StackOverflow](https://stackoverflow.com/questions/52978096/json-b-hitting-npe-serializing-with-adapter-though-setting-withnullvaluestrue).

Added a recreate project w/ instructions [here](https://github.com/scottkurz/yasson-135-recreate).

**Comment by aguibert** [comment]
this issue has been fixed by PR #171, closing this issue
